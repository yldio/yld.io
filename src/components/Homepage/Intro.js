import React from 'react'
import styled from 'styled-components'
import remcalc from 'remcalc'
import breakpoint from 'styled-components-breakpoint'
import ReactMarkdown from 'react-markdown'

import illustrationDesktop from '../../images/hero_image_desktop.svg'
import illustrationMobile from '../../images/hero_image_mobile.svg'
import { Grid, Row, Col } from '../grid'
import BlueBackground from '../Common/BlueBackground'
import { SectionTitle, CardTitle, Subtitle } from '../Typography'

const StyledBlueBackground = styled(BlueBackground)`
  margin-top: -${remcalc(36)};
  position: relative;
`

const IntroRow = styled(Row)`
  z-index: 1;
  padding-top: ${({ theme }) => theme.space[4]};
  position: relative;

  ${breakpoint('smallTablet')`
    padding-top: ${({ theme }) => theme.space[6]};
    padding-bottom: ${({ theme }) => theme.space[8]};
  `}

  ${breakpoint('tablet')`
    padding-top: ${({ theme }) => theme.space[6]};
    padding-bottom: ${({ theme }) => theme.space[7]};
  `}

  ${breakpoint('desktop')`
    padding-top: ${({ theme }) => theme.space[8]};
    padding-bottom: ${({ theme }) => theme.space[9]};
  `}
`

const StyledCardTitle = styled(CardTitle)`
  > li {
    list-style: none;
    padding-bottom: ${remcalc(8)};
  }
`

const IntroSectionTitleWrapper = styled.div`
  max-width: ${remcalc(425)};
  padding-bottom: ${({ theme }) => theme.space[4]};

  ${breakpoint('tablet')`
    max-width: ${remcalc(490)};
  `}

  ${breakpoint('desktop')`
    max-width: ${remcalc(550)};
  `}
`

const IntroImageWrapper = styled.div`
  width: 100%;

  ${breakpoint('smallPhone', 'smallTablet')`
    width: 135%;
    margin-top: -14%;
  `}

  ${breakpoint('smallTablet')`
    bottom: 0;
    left: 46.7%;
    right: 0;
    position: absolute;
    width: 1000px;
  `}

  ${breakpoint('tablet')`
    width: 1120px;
  `}

  ${breakpoint('desktop')`
    width: 1500px;
  `}
  `

const IntroImageDesktop = styled.div`
  display: none;
  position: relative;

  ${breakpoint('smallTablet')`
    left: calc(-50% + 127px);
    display: block;
  `}

  ${breakpoint('tablet')`
    left: calc(-50% + 80px);
  `}

  ${breakpoint('desktop')`
    left: calc(-50% + 175px);
  `}

  > img {
    display: block;
  }
`

const IntroImageMobile = styled.div`
  position: relative;
  ${breakpoint('smallTablet')`
    display: none;
  `}

  > img {
    display: block;
  }
`

const StyledSectionTitle = styled(SectionTitle)`
  ${breakpoint('tablet')`
    font-size: ${remcalc(36)};
    line-height: ${remcalc(42)};
  `};

  ${breakpoint('desktop')`
    font-size: ${remcalc(42)};
    line-height: ${remcalc(48)};
  `};
`

const IntroSection = ({ introHeader, introContent: { introContent } }) => {
  return (
    <StyledBlueBackground>
      <Grid>
        <IntroRow>
          <Col width={[1]} style={{ position: 'relative' }}>
            <IntroSectionTitleWrapper>
              <StyledSectionTitle reverse as="h1">
                {introHeader}
              </StyledSectionTitle>
            </IntroSectionTitleWrapper>
            <ReactMarkdown
              renderers={{
                // eslint-disable-next-line
                heading: props => (
                  <Subtitle noPadding reverse muted {...props} />
                ),
                // eslint-disable-next-line
                list: props => <StyledCardTitle reverse {...props} />
              }}
              source={introContent}
            />
          </Col>
        </IntroRow>
      </Grid>
      <IntroImageWrapper>
        {illustrationDesktop && (
          <IntroImageDesktop>
            <img
              alt="YLD homepage illustration for desktop views"
              src={illustrationDesktop}
            />
          </IntroImageDesktop>
        )}
        {illustrationMobile && (
          <IntroImageMobile>
            <img
              alt="YLD homepage illustration for mobile views"
              src={illustrationMobile}
            />
          </IntroImageMobile>
        )}
      </IntroImageWrapper>
    </StyledBlueBackground>
  )
}

export default IntroSection
