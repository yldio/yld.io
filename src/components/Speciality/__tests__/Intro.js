import React from 'react'
import renderComponent from '../../../utils/tests/renderComponent'
import Intro from '../Intro'

describe('Intro', () => {
  const mockSpeciality = {
    title: 'Node.js',
    seoText:
      'Node.js is the fastest growing open source platform in the world. It empowers us to increase our iteration speed and the quality of our work, using a known language that underpins the entire web: JavaScript.',
    introGraphicTitle: 'node',
    introGraphicFile: {},
    introTitle: 'Why use Node.js?',
    introTextTitle1: 'Performance',
    introTextBody1:
      'Node.js allows us to manage high loads, making it perfect for handling peak demand.',
    introTextTitle2: 'Customer experience',
    introTextBody2:
      'With Node.js we can manage very complex API layers and offer a seamless customer experience.',
    introTextTitle3: 'Productivity',
    introTextBody3:
      'Developers love using Node.js and thatâ€™s reflected directly in their productivity and job satisfaction.'
  }
  it('renders correctly', () => {
    const tree = renderComponent(<Intro speciality={mockSpeciality} />)

    expect(tree).toMatchSnapshot()
  })
})
