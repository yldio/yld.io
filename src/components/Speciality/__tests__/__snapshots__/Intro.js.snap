// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders correctly 1`] = `
<section
  className="sc-jTzLTM kEjvE"
>
  <div
    className="sc-jzJRlG joTOfB"
  >
    <div
      className="sc-htpNat toUCL"
    >
      <div
        className="sc-cSHVUG kneMEB"
      >
        <div
          className="sc-bxivhb bNSQFJ"
          width={
            Array [
              1,
              1,
              1,
              1,
              0.5,
            ]
          }
        >
          <div
            className="flex-citwDR iycZHW"
          >
            <h1
              className="sc-EHOje kbeHcN"
            >
              Node.js
            </h1>
            <p
              className="sc-iwsKbI jZrHPG"
              muted={true}
            >
              Node.js is the fastest growing open source platform in the world. It empowers us to increase our iteration speed and the quality of our work, using a known language that underpins the entire web: JavaScript.
            </p>
          </div>
        </div>
        <div
          className="sc-bxivhb bNSQFJ"
          width={
            Array [
              1,
              1,
              1,
              1,
              0.5,
            ]
          }
        >
          <img
            alt="node"
            className="sc-gqjmRU kBVRHJ"
            style={
              Object {
                "maxHeight": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="sc-ifAKCX dJZmHe"
      >
        <div
          className="sc-bxivhb dabrRZ"
          width={
            Array [
              1,
            ]
          }
        >
          <div
            className="sc-bwzfXH eXQzNB"
          >
            <h3
              className="sc-gzVnrw ISMEH"
            >
              Why use Node.js?
            </h3>
          </div>
        </div>
      </div>
      <div
        className="sc-bxivhb dabrRZ"
        width={
          Array [
            1,
          ]
        }
      >
        <div
          className="sc-ifAKCX dJZmHe"
        >
          <div
            className="sc-VigVT lamdPP sc-bxivhb ekqGfg"
            width={
              Array [
                1,
                1,
                1,
                1,
                0.3333333333333333,
              ]
            }
          >
            <div
              className="sc-fjdhpX ftlTcS"
            >
              <h3
                className="sc-htoDjs bPZptn"
              >
                Performance
              </h3>
              <p
                className="sc-iwsKbI cfIwHc"
                muted={true}
              >
                Node.js allows us to manage high loads, making it perfect for handling peak demand.
              </p>
            </div>
          </div>
          <div
            className="sc-VigVT lamdPP sc-bxivhb ekqGfg"
            width={
              Array [
                1,
                1,
                1,
                1,
                0.3333333333333333,
              ]
            }
          >
            <div
              className="sc-fjdhpX ftlTcS"
            >
              <h3
                className="sc-htoDjs bPZptn"
              >
                Customer experience
              </h3>
              <p
                className="sc-iwsKbI cfIwHc"
                muted={true}
              >
                With Node.js we can manage very complex API layers and offer a seamless customer experience.
              </p>
            </div>
          </div>
          <div
            className="sc-VigVT lamdPP sc-bxivhb ekqGfg"
            width={
              Array [
                1,
                1,
                1,
                1,
                0.3333333333333333,
              ]
            }
          >
            <div
              className="sc-fjdhpX ftlTcS"
            >
              <h3
                className="sc-htoDjs bPZptn"
              >
                Productivity
              </h3>
              <p
                className="sc-iwsKbI cfIwHc"
                muted={true}
              >
                Developers love using Node.js and thatâ€™s reflected directly in their productivity and job satisfaction.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
